workflow/executor/agent.go

```diff
@@ -124,7 +124,11 @@ func (ae *AgentExecutor) taskWorker(ctx context.Context, taskQueue chan task, re
 		result, err := ae.processTask(ctx, tmpl)
 		if err != nil {
 			log.WithError(err).Error("Error in agent task")
-			return
```

Comment:
If we return here, the task worker will be stopped without a restart and the error will never be propagated. This can leave the agent with no task worker threads, deadlocking workflow execution